### DISCUSSION TITLE
consistency of "block_no"

### DISCUSSION THREAD

**Discussion by leoshine** [pymupdf/PyMuPDF#949] [Q&A] [answered]
Dear Authors,

First, I would like to thank you for your great contribution.

My question is:

Both of the following two sets of functions return "block_no". 
However, I find for a same "word" in the PDF, the returned "block_no" are different.
So I'm wondering is this intentional?

1)  page.getText("blocks") / extractBLOCKS():
     >> (x0, y0, x1, y1, "lines in blocks", block_no, block_type)

2)  page.getText("words")  / extractWORDS():
      >> (x0, y0, x1, y1, "word", block_no, line_no, word_no)

Best,
Leo

**Comment by JorjMcKie** [comment, marked as answer]
Not sure I understand:
The same word occurring several times on a page is not unusual or is it? It should have different coordinates and a different combination of block/line/word number each time then.

**Reply by JorjMcKie** [reply]
Ah, and the block number in the "words" list should also be the same as the block number in the "blocks" list, wherethe word occurs.

**Reply by leoshine** [reply]
Hi JorjMcKie,

Thanks for your swift reply.
I also think the block number should be the same in both cases.
I will double-check my code to see if I can find an example, or more likely, find a bug in my code :)

Best,
Leo

**Reply by JorjMcKie** [reply]
Presuming you did **_not_** find any inconsistency  ...
